src/build_options.rs

```diff
@@ -91,6 +91,10 @@ pub struct BuildOptions {
     /// Only applies to macOS targets, do nothing otherwise.
     #[clap(long)]
     pub universal2: bool,
+    /// The `--profile` to pass to cargo. Can also be set by the `MATURIN_PROFILE` environment
+    /// variable.
+    #[clap(long)]
```

Comment:
clap supports [`env`](https://docs.rs/clap/latest/clap/struct.Arg.html#method.env):

```suggestion
    #[clap(long, env = "MATURIN_PROFILE")]
```